--- rstudio-0.99.39/src/cpp/core/include/core/FilePath.hpp~	2015-02-19 21:35:05.000000000 +0100
+++ rstudio-0.99.39/src/cpp/core/include/core/FilePath.hpp	2015-02-19 21:35:13.360551778 +0100
@@ -23,7 +23,9 @@
 #include <boost/scoped_ptr.hpp>
 #include <boost/function.hpp>
 
+#ifndef Q_MOC_RUN
 #include <boost/utility.hpp>
+#endif
 
 #include <core/Error.hpp>
 #include <core/Log.hpp>
--- rstudio-0.99.39/src/cpp/desktop/DesktopSessionLauncher.hpp~	2015-02-19 21:37:22.000000000 +0100
+++ rstudio-0.99.39/src/cpp/desktop/DesktopSessionLauncher.hpp	2015-02-19 21:46:51.487213390 +0100
@@ -18,7 +18,9 @@
 
 #include <string>
 
+#ifndef Q_MOC_RUN
 #include <boost/utility.hpp>
+#endif
 
 #include <core/Error.hpp>
 #include <core/FilePath.hpp>
